### ISSUE TITLE
False positive Useless phpcsSuppress errors

### ISSUE THREAD

**Issue by arxeiss** [slevomat/coding-standard#1226]
I'm using PHP `7.4` and `7.0.8` of this coding standards. I have this code below and I'm getting error `Useless @phpcsSuppress SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint`.

```php
class RegisterController extends Controller
{
    use RegistersUsers;

    /**
     * The user has been registered.
     *
     * @param         User $user
     * @return        mixed
     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint
     */
    protected function registered(Request $request, $user)
    {
        return \redirect()->intended($this->redirectPath());
    }
}
```

 But if I remove those suppress tags, I'm getting error that native typehint can be used. Which cannot be used due to used trait. Do I miss something? Or is it a bug?

**Comment by kukulich** [comment]
It’s not a bug. PHPCS works in one-file scope. It does not know anything about the trait.

**Comment by kukulich** [comment]
It looks like a bug.

**Comment by arxeiss** [comment]
I tried to remove the line with @phpcsSuppress for **Parametr**. So now I'm getting those errors:

```
 81 | ERROR | [x] Method \SunApp\Http\Controllers\Auth\RegisterController::registered() does not have native type hint for its parameter $user but it should be possible to add
    |       |     it based on @param annotation "User". (SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint)
 81 | ERROR | [x] Useless @phpcsSuppress SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint
    |       |     (SlevomatCodingStandard.TypeHints.ReturnTypeHint.UselessSuppress)

```

**Comment by arxeiss** [comment]
I don't know what was wrong, but it started to work. Not sure if there is a cache or something.

Closing until I will be able to simulate it again.
